src/app/shared/navigation-focus/navigation-focus.ts

```diff
@@ -1,24 +1,57 @@
-import {NgModule, OnInit, Directive, ElementRef, HostBinding} from '@angular/core';
+import {NgModule, OnInit, Directive, ElementRef, HostBinding, OnDestroy} from '@angular/core';
+import {Event, Router, NavigationEnd, ResolveEnd} from '@angular/router';
+import {filter, map} from 'rxjs/operators';
+import {Subscription} from "rxjs";
 
 /** The timeout id of the previous focus change. */
 let lastTimeoutId = -1;
 
 @Directive({
   selector: '[focusOnNavigation]',
 })
-export class NavigationFocus implements OnInit {
+export class NavigationFocus implements OnInit, OnDestroy {
   @HostBinding('tabindex') role = '-1';
 
-  constructor(private el: ElementRef) {}
+  private softNavFocus = false;
+  private subscriptions = new Subscription();
+
+  constructor(private el: ElementRef, private router: Router) {
+    // subscribing to events in constructor to catch the correct NavigationEnd/ResolveEnd from previous page
+    // and determine whether the element in question needs to be focused
+    // switching between different components/cdk produces a ResolveEnd instead of NavigationEnd event
+    // the focus is set in ngOnInit to account for late loaded content
+    this.subscriptions.add(
+      this.router.events
+        .pipe(
+          filter((event: Event) => event instanceof NavigationEnd || event instanceof ResolveEnd),
+          map(navEnd => navEnd as NavigationEnd))
```

Comment:
This map is unnecessary, it's just serving as an expensive type cast. Also it seems like maybe a lie since it could be a `ResolveEnd` - why do we want `ResolveEnd` in there? That seems not right

You could maybe try using this alternate signature of filter (notice 2 type parameters):
```ts
function filter<T, S extends T>(predicate: (value: T, index: number) => value is S,
                                       thisArg?: any): OperatorFunction<T, S>;
```

In the context of your code it would look something like:
```ts
filter<RouterEvent, NavigationEnd>(...)
```

Reply:
Removed map and changed to just using `ResolveEnd`